docs/dev/allocation.qmd

```diff
@@ -0,0 +1,268 @@
+
+```julia
+problem[:F] = JuMP.@variable(problem, F[edge = edges] >= 0.0)
+```
+
+More details about setting up variables in allocation can be found in the section [below](https://deltares.github.io/Ribasim/dev/allocation.html#optimization-problem).
+
+- Constraints: These are the constraints that the optimization variables must satisfy.
+They are defined using the [`@constraint`](https://jump.dev/JuMP.jl/stable/api/JuMP/#@constraints) macro.
+The definition of the edge capacity constraints is shown in section [below](https://deltares.github.io/Ribasim/dev/allocation.html#setting-the-constraints-and-capacities).
+`add_constraints_...` functions are used to [add constraints](https://deltares.github.io/Ribasim/dev/allocation.html#setting-up-initial-optimization-constraints) to the optimization problem.
+The [initial value of the constraints](https://deltares.github.io/Ribasim/dev/allocation.html#setting-the-constraints-and-capacities) is set in the function `set_initial_values_...`.
+During the iteration, the constraints are updated based on the current state of the allocation network.
+When [looping over priorities](https://deltares.github.io/Ribasim/dev/allocation.html#updating-capacities), the constraints are updated by the function `adjust_...`.
+
+- Objective function: This is the function that sets the objective of the optimization problem.
+It is defined using the [`@objective`](https://jump.dev/JuMP.jl/stable/api/JuMP/#@objective) macro.
+
+The functions `JuMP.normalized_rhs` and `JuMP.set_normalized_rhs` are used to read and write the constant right hand side of constriants.
+
+For example, to update the capacity of one of the edges, `JuMP.normalized_rhs` moves all the constants to the right-hand sides and all variables to the left-hand side and `JuMP.set_normalized_rhs` sets the new right-hand-side value.
+```julia
+JuMP.set_normalized_rhs(
+    constraints_capacity[edge_id],
+    JuMP.normalized_rhs(constraints_capacity[edge_id]) - JuMP.value(F[edge_id]),
+)
+```
+
+Some JuMP data structures are used to store intermediate or result data. For more information, see [JuMP API](https://jump.dev/JuMP.jl/stable/api/JuMP/).
+
+# Initialization
+Initialization of the allocation data structures happens in `allocation_init.jl`. Below the steps of allocation problem initialization are explained.
+
+For each subnetwork, an allocation problem is made, which is stored in the `allocation_models` field mentioned above.
+
+## Data processing
+### Deriving edge capacities
+
+Edge capacities are important constraints in the optimization problem. Obtaining the capacities of all the flow edges in the subnetworks. The capacity of an edge is given by the smallest `max_flow_rate` of the nodes connected to the edges if these nodes have such a value. The capacities are stored in a `SparseArray` object from `JuMP.jl` called `capacities`, indexed by a tuple of node IDs.
```

Comment:
the following sentence is not correct grammar. I do not understand what is meant. Obtaining the capacities of all the flow edges in the subnetworks.